eips_code/eipXXXX.py

```diff
@@ -0,0 +1,88 @@
+from dataclasses import dataclass
+from eipXXXX_table import TABLE, OP_RJUMP, OP_RJUMPI, OP_CALLF, OP_RETF
+from eip4750 import validate_code_section, immediate_sizes
+
+@dataclass
+class FunctionType:
+    inputs: int
+    outputs: int
+
+
+class ValidationException(Exception):
+    pass
+
+
+def validate_function(func_id: int, code: bytes, types: list[FunctionType] = [FunctionType(0, 0)]) -> int:
+    assert func_id >= 0
+    assert types[func_id].inputs >= 0
+    assert types[func_id].outputs >= 0
+
+    validate_code_section(code, len(types))
+
+    stack_heights = {}
+    start_stack_height = types[func_id].inputs
+    max_stack_height = start_stack_height
+
+    # queue of instructions to analyze, list of (pos, stack_height) pairs
+    worklist = [(0, start_stack_height)]
+
+    while worklist:
+        pos, stack_height = worklist.pop(0)
+        while True:
+            # Assuming code ends with a terminating instruction due to previous validation in validate_code_section()
+            assert pos < len(code), "code is invalid" 
+            op = code[pos]
+            info = TABLE[op]
+
+            # Check if stack height (type arity) at given position is the same
+            # for all control flow paths reaching this position.
+            if pos in stack_heights:
+                if stack_height != stack_heights[pos]:
+                    raise ValidationException("Stack height mismatch for different paths.")
+                else:
+                    break
+            else:
+                stack_heights[pos] = stack_height
+
+
+            stack_height_required = info.stack_height_required
+            stack_height_change = info.stack_height_change
+
+            if op == OP_CALLF:
+                called_func_id = int.from_bytes(code[pos + 1:pos + 3], byteorder="big", signed=False)
+                # Assuming called_func_id is valid due to previous validation in validate_code_section()
+                stack_height_required += types[called_func_id].inputs
+                stack_height_change += types[called_func_id].outputs - types[called_func_id].inputs
+
+            # Detect stack underflow
+            if stack_height < stack_height_required:
+                raise ValidationException("Stack underflow.")
```

Comment:
```suggestion
                raise ValidationException("stack underflow")
```
Don't use full statement?